Create placeholder loading page

gcsecsey opened the pull request:
#### Changes proposed in this Pull Request

Show placeholders to the plugin details page while data is fetched.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Go to plugins page
- Select a plugin
- Check that the loading state has a design matching below:

| |Before | After|
|---|-------|------|
| Desktop | ![image](https://user-images.githubusercontent.com/11555574/142018466-86a98ea3-1c5f-4399-99a4-74c2aac84acc.png) | ![image](https://user-images.githubusercontent.com/11555574/142019033-58434c29-5f08-4b27-b7f6-698c94efc4f8.png) |
| Mobile | ![image](https://user-images.githubusercontent.com/11555574/142018581-dcbc01c1-a5f1-4510-9c30-a58551e7d8e6.png)| ![image](https://user-images.githubusercontent.com/11555574/142018889-85cd6774-1504-47ec-9b0c-c4d4a6a27b7a.png) |

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #57748

gcsecsey committed:
add placeholder styling to plugin page components

gcsecsey committed:
update styling

gcsecsey committed:
add placeholder elements

WBerredo reviewed (dismissed):
I would suggest also adding `user-select: none` to placeholder css, as currently the user can still select and display the reticences inside. Demo:

![Screen Shot 2021-11-16 at 14 49 19](https://user-images.githubusercontent.com/5039531/142047285-66d0a597-bffc-477e-9efd-a26da9087e4a.png)

cpapazoglou commented on client/my-sites/plugins/style.scss:
```diff
@@ -160,6 +161,13 @@ $plugin-details-header-padding: 100px;
 	}
 }
 
+%placeholder {
+	cursor: default;
+	color: transparent;
+	background-color: var( --color-neutral-0 );
+	animation: loading-fade 1.6s ease-in-out infinite;
+}
+
```
TIL about the placeholder selector
https://sass-lang.com/documentation/style-rules/placeholder-selectors

WBerredo replied:
+1

gcsecsey replied:
+1 while working on this! 😄 I didn't want to copy that same CSS into two rules, and I knew that Sass has some mechanism for this use case.

cpapazoglou reviewed (commented):
Works well and looks good. Addressing @WBerredo comment should be enough for merging this.

Personal taste:
I would omit the bottom border
![SS 2021-11-17 at 10 57 06](https://user-images.githubusercontent.com/12430020/142169137-77f2947f-b940-4062-bb1e-a1b497bdc63d.png)

gcsecsey committed:
remove horizontal separator on placeholder

update styles while keeping specificity order

gcsecsey commented:
- Added `user-selection: none` to the CSS, so the placeholder content is no longer selectable (<kbd>Cmd</kbd>+<kbd>A</kbd> won't select it either)
- Removed the horizontal separator on the placeholder
![image](https://user-images.githubusercontent.com/11555574/142235930-8e6b5e35-1b4d-45b6-aaa1-26134be6cbf7.png)

cpapazoglou reviewed (approved):
LGTM!
